index.html

```diff
@@ -1177,119 +1177,305 @@ <h2 id="relationship-to-permissions-policy">
+              Definition
+            </h5>
+            <p>
+              [=remote end definition=]
+            </p>
+            <pre class="cddl remote-cddl">
+              BrowserCommand = (
+                browser.setPermission
+              )
+            </pre>
+          </section>
+          <section>
+            <h5 id="extension-module-browser-types">
+              Types
+            </h5>
+            <section>
+              <h6 id="extension-module-type-browser-PermissionDescriptor">
+                The browser.PermissionDescriptor Type
+              </h6>
+              <pre class="cddl local-cddl">
+                browser.PermissionDescriptor = {
+                  type: "permissionDescriptor",
+                  name: text,
+                }
+              </pre>
+              <p>
+                The <code>browser.PermissionDescriptor</code> type represents a {{PermissionDescriptor}}.
+              </p>
+            </section>
+            <section>
+              <h6 id="extension-module-type-browser-PermissionState">
+                The browser.PermissionState Type
+              </h6>
+              <pre class="cddl local-cddl">
+                browser.PermissionState = "granted" / "denied" / "prompt"
+              </pre>
+              <p>
+                The <code>browser.PermissionState</code> type represents a {{PermissionState}}.
+              </p>
+            </section>
+          </section>
+          <section>
+            <h5 id="extension-module-browser-commands">
+              Commands
+            </h5>
+            <section>
+              <h6 id="extension-module-command-browser-setPermission">
+                The browser.setPermission Command
+              </h6>
+              <p>
+                The <dfn class="export" data-dfn-for="extension modules">Set Permission</dfn>
+                [=extension module=] simulates user modification of a {{PermissionDescriptor}}'s
+                [=permission state=].
+              </p>
+              <dl>
+                <dt>Command Type</dt>
+                <dd>
+                  <pre class="cddl remote-cddl">
+                    browser.setPermission = (
+                      method: "browser.setPermission",
+                      params: browser.SetPermissionParameters
+                    )
+
+                    browser.SetPermissionParameters = {
+                      descriptor: browser.PermissionDescriptor,
+                      state: browser.PermissionState,
+                      ? context: browsingContext.BrowsingContext,
+                      ? origin: text,
+                    }
+                  </pre>
+                </dd>
+                <dt>Result Type</dt>
+                <dd>
+                  <pre class="cddl">
+                    EmptyResult
+                  </pre>
+                </dd>
+             </dl>
+             <div algorithm="remote end steps for permissions.setPermission">
```

Comment:
would it make sense to extract the common algorithm for actually setting the permissions? we probably do not want to just duplicate the setting steps

Reply:
But the variables are different in BiDi and classic. How do we "bind" them in the same algorithm?

Reply:
So looks like the bidi steps 6 to 10 and classic steps 5 to 9 are the same and could be extracted into an algorithm

Reply:
+1, you can either create a new variable to capture `parameters.state` somewhere before step 4 (in classic), and pass it into an algorithm with a better name than `to 'foo bar' with state`, or just pass in `parameters.state` directly to said algorithm.

Reply:
Done.